catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/RemoteDeleteOperations.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.impl.operations;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.Constants;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.impl.FrameworkProperties;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.operation.impl.DeleteResponseImpl;
+import ddf.catalog.operation.impl.ProcessingDetailsImpl;
+import ddf.catalog.source.CatalogStore;
+import ddf.catalog.source.IngestException;
+
+public class RemoteDeleteOperations {
+
+    static final Logger INGEST_LOGGER = LoggerFactory.getLogger(Constants.INGEST_LOGGER_NAME);
+
+    private OperationsCatalogStoreSupport opsCatStoreSupport;
+
+    private FrameworkProperties frameworkProperties;
+
+    private OperationsMetacardSupport opsMetacardSupport;
+
+    public RemoteDeleteOperations(FrameworkProperties frameworkProperties,
+            OperationsMetacardSupport opsMetacardSupport,
+            OperationsCatalogStoreSupport opsCatStoreSupport) {
+
+        this.opsCatStoreSupport = opsCatStoreSupport;
+        this.frameworkProperties = frameworkProperties;
+        this.opsMetacardSupport = opsMetacardSupport;
+
+    }
+
```

Comment:
If we both accept a null `deleteResponse` (line 66) and can return that parameter (line 62) then we should annotate this method with `@Nullable`. And as mentioned if it accepts a null `deleteResponse` we should probably mark that parameter as `@Nullable` as well.